pkgs/stdenv/generic/check-meta.nix

```diff
@@ -123,7 +123,9 @@ let
   #   allowUnfree = false;
   #   allowUnfreePredicate = (x: pkgs.lib.hasPrefix "vscode" x.name);
   # }
-  allowUnfreePredicate = config.allowUnfreePredicate or (x: false);
+  allowUnfreePredicate =
+    config.allowUnfreePredicate
+      or (pkg: builtins.elem (lib.getName pkg) config.allowUnfreePackages or [ ]);
```

Comment:
What happens when both `config.allowUnfreePackages` and `allowUnfreePredicate` are set? Is that something we'd like to assert against?

Reply:
My current thinking is, that it is a nice feature that I can define my own `allowUnfreePredicate` that ignores or uses `config.allowUnfreePackages` however I see fit.

It would also be possible that a self defined predicate is automatically merged with the packages defined in `config.allowUnfreePackages` - but I do not know if that would be the nix way, given that it basically prevents me from temporarily overriding `config.allowUnfreePackages` by setting my own `allowUnfreePredicate`. I would definitely like some input from experienced nixpkgs developers on this.